client/packages/invoices/src/Returns/api/hooks/document/useUpdateInboundReturn.ts

```diff
@@ -1,12 +1,26 @@
-import { useQueryClient, useMutation } from '@openmsupply-client/common';
+import {
+  useQueryClient,
+  useMutation,
+  useDebounceCallback,
+} from '@openmsupply-client/common';
 import { useReturnsApi } from '../utils/useReturnsApi';
 
+const DEBOUNCE_TIME = 500;
+
 export const useUpdateInboundReturn = () => {
   const queryClient = useQueryClient();
   const api = useReturnsApi();
-  return useMutation(api.updateInboundReturn, {
+  const mutation = useMutation(api.updateInboundReturn, {
     onSuccess: () => {
       return queryClient.invalidateQueries(api.keys.base());
     },
   });
+
+  const debouncedMutateAsync = useDebounceCallback(
```

Comment:
I kind of like thise debounceMutate, I think if it also kept track of state, would be pretty useful